src/schemas/gqlTypeDefs.ts

```diff
@@ -339,6 +354,8 @@ const typeDefs = gql`
     oldValue: String!
     donorId: String!
   }
+
+  scalar SchemaList
```

Comment:
This scalar is a workaround for the issue in `/src/dictionary/api`
`clinicalSubmissionTypesList` cannot return `[String] | [SchemaWithFieldsObject]`; unions cannot include primitives

Reply:
Can you link me to more on this please? I don't understand and my gut is telling me that unions should be ok with primitives.

Reply:
They aren't: https://www.apollographql.com/docs/apollo-server/schema/unions-interfaces/
Only allows Object types
Entire Github thread about it: https://github.com/graphql/graphql-spec/issues/215

Reply:
Going to sleep on it and review in the morning
In Gateway, this is simply `[String]` because the response array is converted to JSON, so that may be what client expects and I should just stick to the same format
I was trying to format this to see the proper object for GQL Server purposes, which I think is ideal but might require client side changes

Reply:
sounds good - let me know where the union is happening tomorrow please - read the docs + github - very interesting - looks like unions don't work on scalars either 
```
All of a union's included types must be [object types](https://www.apollographql.com/docs/apollo-server/schema/schema/#object-types) (not scalars, input types, etc.). Included types do not need to share any fields.
```

Reply:
Oh so this Union is related to getSchemasWithFields: https://github.com/icgc-argo/argo-clinical/pull/1060#discussion_r1333305009
Where were either returning a String[] of Schema names or an array of SchemaWithFields objects